junit-async/native/src/main/scala/scala/scalanative/junit/async/package.scala

```diff
@@ -0,0 +1,12 @@
+package scala.scalanative.junit
+
+import scala.concurrent.Future
+import scala.util.Try
+
+package object async {
+  type AsyncResult = Try[_]
```

Comment:
This should be `Unit`, since that is the only result type supported by the JUnit plugin.
```suggestion
  type AsyncResult = Unit
```
Currently `await` returns a `Try`, but `@Test` methods calling `await` are declared to return `Unit`. That means that the `Try` is discarded, hence any exception caught in the `Try` is discarded!